### PR TITLE
Feat/ Integrate cypress test and coverage to GHA

### LINKED ISSUE #360: Generate Cypress coverage

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Fixes #360 

Enables Cypress coverage generation and having the result uploaded to sonar cloud.

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-541-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-541-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-541-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-541-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-541-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-541-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: .github/workflows/pr-open.yml [hunk 1]
```diff
@@ -172,20 +169,20 @@ jobs:
           working-directory: ./frontend
           browser: chrome
 
-      - name: Upload screenshots on failure
-        uses: actions/upload-artifact@v3
-        if: failure()
-        with:
-          name: cypress-screenshots-chrome
-          path: /home/runner/work/nr-spar/nr-spar/frontend/cypress/screenshots
```
Comment (mgaseta) [inline .github/workflows/pr-open.yml:180]: Will we stop uploading failure's data?

File: .github/workflows/pr-open.yml [hunk 2]
```diff
@@ -42,8 +42,7 @@ jobs:
           oc_token: ${{ secrets.OC_TOKEN }}
           file: common/openshift.init.yml
           overwrite: false
-          parameters:
-            -p ZONE=${{ github.event.number }}
+          parameters: -p ZONE=${{ github.event.number }}
```
Comment (mgaseta) [inline .github/workflows/pr-open.yml:45]: Why this change? It doesn't make sense (for me) to put only this parameter on this line. I'm adding only this comment but the same goes for similar changes on this file.

**Review by mgaseta — CHANGES_REQUESTED** [review]
LGTM! :1st_place_medal: 
Added only a couple of question.

File: .github/workflows/pr-open.yml (hunk 1 above)
Comment (craigyu) [reply .github/workflows/pr-open.yml:180]: We upload videos on each run so the screenshots is quite useless

**Comment by craigyu** [comment]
image

Coverage and dup detection seem to be working

File: .github/workflows/pr-open.yml (hunk 2 above)
Comment (craigyu) [reply .github/workflows/pr-open.yml]: fixed here https://github.com/bcgov/nr-spar/pull/541/commits/c3e0abd6ab092c2ef9740756f88be47259d4861e

**Review by RMCampos — APPROVED** [review]
LGTM

**Review by mgaseta — APPROVED** [review]
LGTM! :100:

**Pull request merged**